.github/workflows/e2e.yml

```diff
@@ -0,0 +1,81 @@
+name: End-to-end Tests
+on:
+  pull_request:
+    paths:
+      - '*.go'
+      - 'go.mod'
+      - 'go.sum'
+      - 'pkg/**'
+      - 'tools/**'
+      - 'config/**'
+      - '!config/stack/ttn-lw-stack.yml'
+      - 'package.json'
+      - 'pkg/webui/**'
+      - 'sdk/js/**'
+      - 'yarn.lock'
+
+jobs:
+  end-to-end:
+    name: Frontend based (cypress)
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Get Yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(npx yarn cache dir)"
+      - name: Initialize Yarn module cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Initialize messages and locales cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            .cache/babel
+            .cache/messages
+            pkg/webui/console/locales/.backend
+          key: ${{ runner.os }}-messages-cache-${{ hashFiles('pkg/webui/**/*.js') }}-${{ hashFiles('config/messages.json')}}
+      - name: Initialize public folder cache
+        uses: actions/cache@v2
+        with:
+          path: public
+          key: ${{ runner.os }}-public-cache-${{ hashFiles('pkg/webui/**') }}-${{ hashFiles('sdk/js/**/*.js') }}-${{ hashFiles('config/webpack.config.babel.js') }}-${{ hashFiles('**/yarn.lock') }}-${{ hashFiles('pkg/ttnpb/field_mask_validation.go') }}
+      - name: Initialize JS SDK cache
```

Comment:
Why is this only performed here and not in other workflows using JS SDK?
Please add to all relevant workflows in order to avoid inconsistencies. (can be in a separate PR)

Reply:
Is there any action taken in response to this?
Are you going to update this PR? Are you going to open another PR before merging this one? Are you going to file an issue?

Reply:
I removed caching steps that do no appear in other js workflow. We can add those later if we experience any noticeable performance issues.

Reply:
Note that I was able to bring the run time down to about 5 minutes under optimal circumstances with the caching I had applied in the other e2e PR.